app/src/main/java/org/mozilla/fenix/settings/advanced/DefaultLocaleSettingsController.kt

```diff
@@ -16,32 +16,44 @@ interface LocaleSettingsController {
 }
 
 class DefaultLocaleSettingsController(
-    private val context: Context,
+    private val activity: Activity,
     private val localeSettingsStore: LocaleSettingsStore
 ) : LocaleSettingsController {
 
     override fun handleLocaleSelected(locale: Locale) {
         if (localeSettingsStore.state.selectedLocale == locale &&
-            !LocaleManager.isDefaultLocaleSelected(context)) {
+            !LocaleManager.isDefaultLocaleSelected(activity)) {
             return
         }
         localeSettingsStore.dispatch(LocaleSettingsAction.Select(locale))
-        LocaleManager.setNewLocale(context, locale.toLanguageTag())
-        LocaleManager.updateBaseConfiguration(context, locale)
-        (context as Activity).recreate()
+        LocaleManager.setNewLocale(activity, locale.toLanguageTag())
+        LocaleManager.updateBaseConfiguration(activity, locale)
+        activity.recreate()
     }
 
     override fun handleDefaultLocaleSelected() {
-        if (LocaleManager.isDefaultLocaleSelected(context)) {
+        if (LocaleManager.isDefaultLocaleSelected(activity)) {
             return
         }
         localeSettingsStore.dispatch(LocaleSettingsAction.Select(localeSettingsStore.state.localeList[0]))
-        LocaleManager.resetToSystemDefault(context)
-        LocaleManager.updateBaseConfiguration(context, localeSettingsStore.state.localeList[0])
-        (context as Activity).recreate()
+        LocaleManager.resetToSystemDefault(activity)
+        LocaleManager.updateBaseConfiguration(activity, localeSettingsStore.state.localeList[0])
+        activity.recreate()
     }
 
     override fun handleSearchQueryTyped(query: String) {
         localeSettingsStore.dispatch(LocaleSettingsAction.Search(query))
     }
+
+    /**
+     * Update the locale for the configuration of the app context's resources
+     */
+    @Suppress("Deprecation")
+    fun LocaleManager.updateBaseConfiguration(context: Context, locale: Locale) {
```

Comment:
is there an alternative to this deprecated method?

Reply:
I'd prefer to leave that as follow up work. I filed #12562